homeassistant/components/switch/wemo.py

```diff
@@ -109,12 +110,33 @@ def device_state_attributes(self):
 
         if self.insight_params or (self.coffeemaker_mode is not None):
             attr[ATTR_CURRENT_STATE_DETAIL] = self.detail_state
+            attr['current_power_w'] = \
```

Comment:
Don't expect  the coffee machine will support these new attributes - so I'd change this so that you only retrieve these attributes for the insight device.